client/src/api/v2/camera.ts

```diff
@@ -0,0 +1,271 @@
+  status: CameraRecordingStatus;
+  /** Current duration of recording in minutes, if recording */
+  recordingMinutes?: number;
+  /** Filename of recording, if recording */
+  recordingFile?: string;
+  /** Remaining disk space available for recording, in bytes */
+  diskSpaceRemaining: number;
+  /** Description of any error that has occurred */
+  error?: string;
+  /** Catch-all for any extra data */
+  [propName: string]: any;
+}
+
+interface CameraRecordingStatusFormatted {
+  /** Status of the camera's recording */
+  status?: string;
+  /** Current duration of recording in mm:ss, if recording */
+  recordingMinutes?: string;
+  /** Filename of recording, if recording */
+  recordingFile?: string;
+  /** Remaining disk space available for recording */
+  diskSpaceRemaining?: string;
+  /** Description of any error that has occurred */
+  error?: string;
+  /** Catch-all for any extra data */
+  [propName: string]: any;
+}
+
+/**
+ * Parse the recording payload into an object and format the values
+ *
+ * Payload structure is defined in the 'V3 MQTT Topics' page on Notion.
+ * Topic is /v3/camera/recording/status/<primary/>secondary>.
+ *
+ * @param data Payload from MQTT message, parsed
+ * @returns Formatted payload
+ */
+function parseRecordingPayloadData(data: CameraRecordingStatusPayload) {
+  if (!data) return null;
+
+  const formattedData: CameraRecordingStatusFormatted = {};
+
+  Object.keys(data).forEach((field) => {
+    let name = camelCaseToStartCase(field);
+    let value = '';
+
+    // format field value
+    switch (field) {
+      case 'status':
+        value = capitalise(data.status);
+        break;
+
+      case 'diskSpaceRemaining':
+        value = formatBytes(data.diskSpaceRemaining);
+        break;
+
+      case 'recordingMinutes':
+        {
+          const totalMinutes = data.recordingMinutes as number;
+          const mins = Math.floor(totalMinutes);
+          const secs = Math.floor((totalMinutes - mins) * 60);
+          value = `${mins}m ${secs}s`;
+          name = 'Recording Time';
+        }
+        break;
+
+      default:
+        value = data[field];
+        break;
+    }
+
+    formattedData[name] = value;
+  });
+
+  return formattedData;
+}
+
+/**
+ * Initiate receiving status payloads
+ *
+ * @param subComponent The sub component to init for (e.g. recording, video feed)
+ * @param device Device
+ */
+function initStatus(subComponent: string, device?: CameraDevice) {
+  const deviceAsArr = device ? [device] : []; // only add device path if specified
+  const path = ['camera', subComponent].concat(deviceAsArr);
+  emit(`get-status-payload`, path);
+}
+
+/**
+ * Makes the device name pretty.
+ *
+ * Hardcoded for efficiency. If you don't like it, complain to Angus Trau :).
+ *
+ * @param device Device
+ * @returns  Prettied device name
+ */
+export function getPrettyDeviceName(device: CameraDevice) {
+  return device === CameraDevice.Primary ? 'Primary' : 'Secondary';
+}
```

Comment:
Isn't the value stored in the `CameraDevice` enum? Is there any reason we can't do `device.toUpperCase()` @angustrau ?

Reply:
This used to be a string literal (either "primary" or "secondary"). We could keep it that way by using a union type in the string literals.

Reply:
@angustrau I would've used string literals too, but I do see the usefulness of enums instead.

Reply:
Is there a benefit of using unioned string literals compared to enums?